WallPaperDev/WallPaperDev/Views/Welcome Views/WelcomeCellViews.swift

```diff
@@ -8,185 +8,156 @@
-    fingerMoveDownUP.toValue = phoneLayer.bounds.maxY
-    fingerMoveDownUP.autoreverses = true
-    fingerMoveDownUP.duration = 0.5
-    textMoveDown.timingFunction = CAMediaTimingFunction(name: .easeInEaseOut)
-    fingerLayer.add(fingerMoveDownUP, forKey: nil)
-  }
-}
-
-class ShowCaseView: UIView {
-  private let headerText = Localized.string("tutorial_showcase_title")
-  private let subHeaderText = Localized.string("tutorial_showcase_message")
-  private let container = CALayer()
-  private let phoneLayer = CALayer()
-  private let headerLabel = WelcomeLabel(frame: .zero)
-  private let subHeaderLabel = WelcomeLabel(frame: .zero)
-  
-  override init(frame: CGRect) {
-    super.init(frame: frame)
-    backgroundColor = .white
-    setupUI()
-    setupLayers()
-  }
-  
-  required init?(coder: NSCoder) {
-    fatalError("init(coder:) has not been implemented")
-  }
-  
-  func resetAnimation() {
-    prepAnimation()
-  }
-  
-  private func setupUI() {
-    subHeaderLabel.configSubHeaderLabel(text: subHeaderText)
-    subHeaderLabel.frame.size.width = bounds.width * 0.7
-    subHeaderLabel.sizeToFit()
-    subHeaderLabel.center = CGPoint(x: center.x, y: bounds.maxY - subHeaderLabel.frame.height / 2)
-    
-    headerLabel.configHeaderLabel(text: headerText)
-    headerLabel.frame.size.width = bounds.width * 0.8
-    headerLabel.sizeToFit()
-    headerLabel.center = center
-    headerLabel.center.y = subHeaderLabel.frame.minY - headerLabel.frame.height / 2
-    
-    addSubview(headerLabel)
-    addSubview(subHeaderLabel)
-    
-    container.frame = CGRect(origin: bounds.origin, size: CGSize(width: bounds.width, height: bounds.height - headerLabel.frame.height - subHeaderLabel.frame.height))
-  }
-  
-  private func setupLayers() {
-    container.frame = CGRect(origin: bounds.origin, size: CGSize(width: bounds.width, height: bounds.height - headerLabel.frame.height - subHeaderLabel.frame.height))
-    layer.addSublayer(container)
-    
-    let phoneSize = CGSize(width: container.bounds.width * 0.6, height: container.bounds.height)
-    let phoneXOffset = bounds.width / 5
-    phoneLayer.frame = CGRect(origin: CGPoint(x: phoneXOffset, y: 0), size: phoneSize)
-    phoneLayer.contents = ApplicationDependency.manager.currentTheme.imageAssets.tutorialPhone.cgImage
-    phoneLayer.contentsGravity = .resizeAspect
-    container.addSublayer(phoneLayer)
-  }
-  
-  private func prepAnimation() {
-    let scrollUp = CABasicAnimation(keyPath: "position.y")
-    scrollUp.fromValue = -phoneLayer.frame.height + 100
-    scrollUp.toValue = CGAffineTransform.identity
-    scrollUp.duration = 0.5
-    scrollUp.timingFunction = CAMediaTimingFunction(name: .easeOut)
-    phoneLayer.add(scrollUp, forKey: nil)
+    layer.addSublayer(fingerLayer)
+  }
+  
+  func animatePhone(for style: WelcomeStyle) {
+    switch style {
+    case .showcase:
+      animateShowcase()
+    case.demo:
+      animateDemo()
+    default:
```

Comment:
One nit, when dealing with enums I find it best to avoid using `default` as if you avoid default it makes future additions of cases require explicit developer checks for the behavior and not just inherit the `default` case

Reply:
I haven't thought about that and I will make that change in the next pull request